api/src/OpenTelemetry/Logging/Core.hs

```diff
@@ -2,29 +2,43 @@
+) where
 
-import Data.Int (Int32, Int64)
+import Data.Int (Int64)
 import Data.Text (Text)
-import OpenTelemetry.Attributes (Attribute)
 import OpenTelemetry.Common
+import OpenTelemetry.LogAttributes (LogAttributes)
 import OpenTelemetry.Resource (MaterializedResources)
 import OpenTelemetry.Trace.Id (SpanId, TraceId)
 
 
-data Log body = Log
+{- | This is a data type that can represent logs from various sources: application log files, machine generated events, system logs, etc. [Specification outlined here.](https://opentelemetry.io/docs/specs/otel/logs/data-model/)
+Existing log formats can be unambiguously mapped to this data type. Reverse mapping from this data type is also possible to the extent that the target log format has equivalent capabilities.
+-}
+data LogRecord body = LogRecord
   { timestamp :: Maybe Timestamp
   -- ^ Time when the event occurred measured by the origin clock. This field is optional, it may be missing if the timestamp is unknown.
+  , observedTimestamp :: Timestamp
+  -- ^ Time when the event was observed by the collection system. For events that originate in OpenTelemetry (e.g. using OpenTelemetry Logging SDK)
+  -- this timestamp is typically set at the generation time and is equal to Timestamp. For events originating externally and collected by OpenTelemetry (e.g. using Collector)
+  -- this is the time when OpenTelemetry’s code observed the event measured by the clock of the OpenTelemetry code. This field SHOULD be set once the event is observed by OpenTelemetry.
+  --
+  -- For converting OpenTelemetry log data to formats that support only one timestamp or when receiving OpenTelemetry log data by recipients that support only one timestamp internally the following logic is recommended:
+  -- - Use Timestamp if it is present, otherwise use ObservedTimestamp
   , tracingDetails :: Maybe (TraceId, SpanId, TraceFlags)
   -- ^ Tuple contains three fields:
   --
   -- - Request trace id as defined in W3C Trace Context. Can be set for logs that are part of request processing and have an assigned trace id.
   -- - Span id. Can be set for logs that are part of a particular processing span.
   -- - Trace flag as defined in W3C Trace Context specification. At the time of writing the specification defines one flag - the SAMPLED flag.
   , severityText :: Maybe Text
-  -- ^ severity text (also known as log level). This is the original string representation of the severity as it is known at the source. If this field is missing and SeverityNumber is present then the short name that corresponds to the SeverityNumber may be used as a substitution. This field is optional.
+  -- ^ severity text (also known as log level). This is the original string representation of the severity as it is known at the source. If this field is missing
+  -- and SeverityNumber is present then the short name that corresponds to the SeverityNumber may be used as a substitution. This field is optional.
   , severityNumber :: Maybe Int64
```

Comment:
There are several possible [ways to represent `severityNumber`](https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitynumber) but `Int64` is the simplest for the moment